src/picongpu/include/particles/ionization/byCollision/ThomasFermi/ThomasFermi_Impl.hpp

```diff
@@ -272,19 +272,33 @@ namespace ionization
                 /* energy density in sim units */
                 float_X kinEnergyDensity = kinEnergyV[0];
 
-                /* this is the point where actual ionization takes place */
+                /* Thomas-Fermi calculation step:
+                 * Determines the new average charge state for each ion under
+                 * LTE conditions. Returns the new number of bound electrons
+                 * according to this average charge state.
+                 */
                 IonizationAlgorithm ionizeAlgo;
                 float_X newBoundElectrons = ionizeAlgo(
                     kinEnergyDensity, density,
                     particle, this->randomGen()
                     );
 
-                particle[boundElectrons_] = newBoundElectrons;
-
-                /* safety check to avoid double counting since recombination is not yet implemented */
+                /* Safety Check: Would the number of bound electrons after the
```

Comment:
this line does not explain why it is save.
as noted in the second paragraph (only changed when ionization occurs), one would expect, that your algorithm above should keep the the variable `newBoundElectrons` to its original value and that this if only guards an necessary write of the same value.

if that is the case, then this comment line makes no sense.
if that is not the case, the `ionizeAlgo` call behaves weird or in an undocumented/unexpected way and needs to be fixed.